tests/transactions_example.rs

```diff
@@ -0,0 +1,91 @@
+#![cfg(all(feature = "tokio-runtime", not(feature = "tokio-sync")))]
+
+// START TRANSACTIONS EXAMPLE
+use mongodb::{
+    bson::{doc, Document},
+    error::{Result, TRANSIENT_TRANSACTION_ERROR, UNKNOWN_TRANSACTION_COMMIT_RESULT},
+    options::{Acknowledgment, ReadConcern, TransactionOptions, WriteConcern},
+    Client,
+    ClientSession,
+};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let uri = std::env::var("MONGODB_URI").expect("MONGODB_URI must be set");
+    let client = Client::with_uri_str(uri).await?;
+
+    let mut session = client.start_session(None).await?;
+    run_transaction_with_retry(&mut session).await
+}
+
+async fn run_transaction_with_retry(session: &mut ClientSession) -> Result<()> {
+    loop {
+        match execute_transaction(session).await {
+            Ok(()) => {
+                println!("Transaction succeeded.");
+                return Ok(());
+            }
+            Err(error) => {
+                if error.contains_label(TRANSIENT_TRANSACTION_ERROR) {
+                    println!("TransientTransactionError, retrying transaction...");
+                    continue;
+                } else {
+                    session.abort_transaction().await?;
+                    return Err(error);
+                }
+            }
+        }
+    }
+}
+
+async fn execute_transaction(session: &mut ClientSession) -> Result<()> {
+    let transaction_options = TransactionOptions::builder()
+        .read_concern(ReadConcern::snapshot())
+        .write_concern(WriteConcern::builder().w(Acknowledgment::Majority).build())
+        .build();
+    session.start_transaction(transaction_options).await?;
+
+    let client = session.client();
+    let employees = client.database("hr").collection::<Document>("employees");
+    let events = client.database("reporting").collection("events");
+
+    employees
```

Comment:
as an aside, this is a perfect use-case for a one-shot transaction in the future 🙂